src/PrestaShopBundle/Controller/Admin/Improve/Modules/AddonsStoreController.php

```diff
@@ -35,28 +36,37 @@
 class AddonsStoreController extends FrameworkBundleAdminController
 {
     /**
-     * @var string the controller name for routing
-     */
-    const CONTROLLER_NAME = 'AdminAddonsCatalog';
-
-    /**
+     *
+     * @AdminSecurity("is_granted(['read'], request.get('_legacy_controller'))")
+     *
      * @param Request $request
      *
      * @return \Symfony\Component\HttpFoundation\Response
      */
     public function indexAction(Request $request)
     {
+        $pageContent = @file_get_contents($this->getAddonsUrl($request));
+
+        if (!$pageContent) {
+            $this->addFlash('error', $this->trans(
+                'An unexpected error occurred.',
```

Comment:
Nice, thanks @matks! I suggest _It looks like this page is unavailable because we have trouble connecting to Addons. Please refresh the page or check your firewall (setup? settings?)._, let me know if it does correspond to the situation. :-)

Reply:
@TristanLDD

Reply:
@matks, let's pick this wording instead: _It looks like we have trouble connecting to Addons. Please refresh the page or check your firewall configuration._